ql/src/java/org/apache/hadoop/hive/ql/exec/MoveTask.java

```diff
@@ -398,11 +398,20 @@ public int execute(DriverContext driverContext) {
         // for transactional table if write id is not set during replication from a cluster with STRICT_MANAGED set
         // to false then set it now.
         if (tbd.getWriteId() <= 0 && AcidUtils.isTransactionalTable(table.getParameters())) {
-          Long writeId = ReplUtils.getMigrationCurrentTblWriteId(conf);
-          if (writeId == null) {
-            throw new HiveException("MoveTask : Write id is not set in the config by open txn task for migration");
+          if (work.getIsInReplicationScope()) {
+            Long writeId = ReplUtils.getMigrationCurrentTblWriteId(conf);
+            if (writeId == null) {
+              throw new HiveException("MoveTask : Write id is not set in the config by open txn task for migration");
+            }
+            tbd.setWriteId(writeId);
+          } else if (tbd.isCtas()) {
```

Comment:
Need not pass isCtas flag. Instead driverContext.getCtx().getHiveTxnManager().getAllocatedTableWriteId can be called and if it returns 0, then throw error.

Reply:
WriteId for the table being created is obtained while creating loadTableDesc itself. So requested change is not required.